MidpointFindingKarel.java

```diff
@@ -1,19 +1,25 @@
-/*
- * File: MidpointFindingKarel.java
- * -------------------------------
- * When you finish writing it, the MidpointFindingKarel class should
- * leave a beeper on the corner closest to the center of 1st Street
- * (or either of the two central corners if 1st Street has an even
- * number of corners).  Karel can put down additional beepers as it
- * looks for the midpoint, but must pick them up again before it
- * stops.  The world may be of any size, but you are allowed to
- * assume that it is at least as tall as it is wide.
- */
-
-import stanford.karel.*;
-
-public class MidpointFindingKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: MidpointFindingKarel.java
+ * -------------------------------
+ * When you finish writing it, the MidpointFindingKarel class should
+ * leave a beeper on the corner closest to the center of 1st Street
+ * (or either of the two central corners if 1st Street has an even
+ * number of corners).  Karel can put down additional beepers as it
+ * looks for the midpoint, but must pick them up again before it
+ * stops.  The world may be of any size, but you are allowed to
+ * assume that it is at least as tall as it is wide.
+ */
+
+import stanford.karel.*;
+
+public class MidpointFindingKarel extends SuperKarel {

```

Comment:
This solution doesn't work given the fact that you are supposed to find a midpoint given different types of worlds.